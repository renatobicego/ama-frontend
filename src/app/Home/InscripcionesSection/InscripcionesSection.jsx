import Image from "next/image";
import Link from "next/link";

const inscripcionesNav = [
    {
        href: '/inscripciones/torneos',
        logo: '/icons/medal.png',
        title: 'Inscripciones a Torneos'
    },
    {
        href: '/inscripciones/federaciones',
        logo: '/icons/track.png',
        title: 'Federaci贸n Anual AMA'
    }
]

const InscripcionesSection = () => {
    return(
        <section className="size-section flex flex-wrap md:justify-between gap-6 lg:gap-[3vw] items-center my-16">
            <h3 className="font-title font-semibold text-title w-52 text-2xl lg:text-3xl">
                Inscripciones Online
            </h3>
            <nav>
                <ul className="flex gap-10 2xl:gap-[3vw] flex-wrap lg:flex-nowrap">

                    {/* Render Federaci贸n and Inscripci贸n a Torneos links */}
                    {inscripcionesNav.map((navItem, i) => {
                        return(
                            <li className="w-full sm:w-1/2 lg:w-full" key={i}>
                                <Link className="flex justify-between items-center gap-4" href={navItem.href}>
                                    {/* Icon left */}
                                    <Image 
                                        width={200} 
                                        height={200} 
                                        src={navItem.logo} 
                                        alt="Icono de inscripciones"
                                        className="max-w-[50px] h-[55px] md:max-w-[70px] md:h-[75px] object-contain border-r-2 pr-4"
                                        />
                                    <h4 className="font-title font-normal text-title text-lg lg:text-xl">
                                        {navItem.title}
                                    </h4>
                                    {/* Arrow icon */}
                                    <button className='cursor-pointer w-1/5'>
                                        <Image 
                                            width={50} 
                                            height={50} 
                                            src='/icons/arrowRight.svg' 
                                            alt="Icono para ir a secci贸n de inscripciones" />
                                    </button>
                                </Link>
                            </li>
                        )
                    })}
                </ul>
            </nav>
            
            
        </section>
    )
}

export default InscripcionesSection