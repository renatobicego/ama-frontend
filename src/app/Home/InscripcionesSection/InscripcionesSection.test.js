import { cleanup, render, screen } from "@testing-library/react"
import InscripcionesSection from "./InscripcionesSection"
import '@testing-library/jest-dom/extend-expect'

describe('<InscripcionesSection />', () => {

    let inscripciones

    beforeEach(() => {
        inscripciones = render(<InscripcionesSection />)
        
    })

    afterEach(() => {
        cleanup()
    })

    test('renders content', () => {
        expect(inscripciones.container).toMatchSnapshot()

        // Render title
        const title = screen.getByText('Inscripciones Online')
        expect(title).toBeInTheDocument()

        // Render icons
        const icons = screen.getAllByAltText(`Icono de inscripciones`)
        expect(icons.length).toBe(2)

        // Render arrow visit section icon
        const arrowIcon = screen.getAllByAltText(`Icono para ir a secci√≥n de inscripciones`)
        expect(arrowIcon.length).toBe(2)

    })
})
