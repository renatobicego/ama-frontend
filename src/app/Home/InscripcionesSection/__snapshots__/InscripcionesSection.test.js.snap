// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InscripcionesSection /> renders content 1`] = `
<div>
  <section
    class="size-section flex flex-wrap md:justify-between gap-6 lg:gap-[3vw] items-center my-16"
  >
    <h3
      class="font-title font-semibold text-title w-52 text-2xl lg:text-3xl"
    >
      Inscripciones Online
    </h3>
    <nav>
      <ul
        class="flex gap-10 2xl:gap-[3vw] flex-wrap lg:flex-nowrap"
      >
        <li
          class="w-full sm:w-1/2 lg:w-full"
        >
          <a
            class="flex justify-between items-center gap-4"
            href="/inscripciones/torneos"
          >
            <img
              alt="Icono de inscripciones"
              class="max-w-[50px] h-[55px] md:max-w-[70px] md:h-[75px] object-contain border-r-2 pr-4"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Ficons%2Fmedal.png&w=640&q=75"
              srcset="/_next/image?url=%2Ficons%2Fmedal.png&w=256&q=75 1x, /_next/image?url=%2Ficons%2Fmedal.png&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <h4
              class="font-title font-normal text-title text-lg lg:text-xl"
            >
              Inscripciones a Torneos
            </h4>
            <button
              class="cursor-pointer w-1/5"
            >
              <img
                alt="Icono para ir a sección de inscripciones"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/icons/arrowRight.svg"
                style="color: transparent;"
                width="50"
              />
            </button>
          </a>
        </li>
        <li
          class="w-full sm:w-1/2 lg:w-full"
        >
          <a
            class="flex justify-between items-center gap-4"
            href="/inscripciones/federaciones"
          >
            <img
              alt="Icono de inscripciones"
              class="max-w-[50px] h-[55px] md:max-w-[70px] md:h-[75px] object-contain border-r-2 pr-4"
              data-nimg="1"
              decoding="async"
              height="200"
              loading="lazy"
              src="/_next/image?url=%2Ficons%2Ftrack.png&w=640&q=75"
              srcset="/_next/image?url=%2Ficons%2Ftrack.png&w=256&q=75 1x, /_next/image?url=%2Ficons%2Ftrack.png&w=640&q=75 2x"
              style="color: transparent;"
              width="200"
            />
            <h4
              class="font-title font-normal text-title text-lg lg:text-xl"
            >
              Federación Anual AMA
            </h4>
            <button
              class="cursor-pointer w-1/5"
            >
              <img
                alt="Icono para ir a sección de inscripciones"
                data-nimg="1"
                decoding="async"
                height="50"
                loading="lazy"
                src="/icons/arrowRight.svg"
                style="color: transparent;"
                width="50"
              />
            </button>
          </a>
        </li>
      </ul>
    </nav>
  </section>
</div>
`;
